docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,167 @@
+	    // This is what allows the app to be responsive and not hang on the UI thread.
+	    var directionsJson = await getDirectionsTask;
+		
+	    // Using the JSON.NET library to deserialize data
+	    var directionsData = JsonConvert.DeserializeObject<DirectionsData>(directionsJson);
+	    		    
+	    DrawRouteOnMap(directionsData.Polyline);
+	}
+
+Web service snippet:
+
+.. code-block:: c#
+
+	[HttpGet]
+	public async Task<string> GetDirections(GeoPoint startPos, GeoPoint endPos)
+	{
+	    // Suspends GetUser() to allow the caller (the web service) to accept another request,
+	    // rather than blocking on this one.
+	    var directions = await _locationService.GetDirectionsAsync(startPos, endPos);
+	    
+	    // Using the JSON.NET library to serialize data
+	    var json = JsonConvert.SerializeObject(directions);
+	    
+	    return json;
+	}
+	
+Bonus snippet: writing an inline event handler in Xamarin for an android game!
+
+.. code-block:: c#
+
+	fireball.DamageDone += async =>
+	{
+	   var result = await DoFireballDamageCalculation();
+	   ShowDamageOnScreen(result);
+	};
+	
+Simple Async Example (VB)
+-------------------------
+
+These are the VB-equivalent code snippets from above.
+
+Client app snippet:
+
+.. code-block:: vb.net
+
+	Private Async Sub GetDirectionsForPickup_Pressed(sender As Object, e As CustomUserDataEventArgs) Handles GetDirectionsForPickup.Click
+		
+		Dim b As BasicUserData = e.Source
+		
+		' _directionsClient is a made-up object that would encapsulate certain complexity,
+		' such as getting the location from the device.
+		Dim getDirectionsTask As Task(Of String) = _directionsClient.GetDrivingDirectionsToLocation(user.LastKnownLocation)
+		
+		' This independent work can be done concurrently
+		' since it doesn't rely on directions!
+		NotifyUserOfPickupAsync(b.ContactInfo)
+		
+		Dim directionsJson As String = Await getDirectionsTask
+		
+		' Using the JSON.NET library to deserialize data
+		Dim d As DirectionsData = JsonConvert.DeserializeObject(Of DirectionsData)(directionsJson)
+		
+		DrawRouteOnMap(d.Polyline)		
+	End Sub
+
+Web Service snippet:
+
+.. code-block:: vb.net
+
+	<HttpGet>
+	Public Async Function GetDirections(startPos as GeoPoint, endPos as GeoPoint) As Task(Of String)
+
+		' Suspends GetUser() to allow the caller (the web service) to accept another request,
+		' rather than blocking on this one.
+		Dim dirs As Directions = Await _locationService.GetDirectionsAsync(startPos, endPost);
+		
+		Dim json As String = JsonConvert.SerializeObject(dirs);
+		
+		Return json
+	End Function
+
+Important Info and Advice
+-------------------------
+
+Although async programming is relatively straightfoward, there are some details to keep in mind which could otherwise result in some nasty behavior.
+
+* Do yourself a favor and append "Async" to the end of every async method you write which could be consumed by another method.
```

Comment:
Maybe formulate it more as a best practice advice? For instance "Best practice is to add an "Async" suffix..." etc.